clippy.toml

```diff
@@ -0,0 +1,5 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+msrv = "1.61.0"
```

Comment:
thought for later: we should also set the MSRV in our [`Cargo.toml`s](https://doc.rust-lang.org/nightly/cargo/reference/manifest.html#the-rust-version-field)